analyzer/src/functTest/java/org/jboss/gm/analyzer/alignment/TestUtils.java

```diff
@@ -24,34 +28,62 @@
     private TestUtils() {
     }
 
+    //TODO use a builder for these parameters
+
     static TestManipulationModel align(File projectRoot, String projectDirName) throws IOException, URISyntaxException {
-        return align(projectRoot, projectDirName, false);
+        return align(projectRoot, projectDirName, new HashMap<>());
+    }
+
+    static TestManipulationModel align(File projectRoot, String projectDirName, Map<String, String> systemProps)
+            throws IOException, URISyntaxException {
+        return align(projectRoot, projectDirName, false, systemProps);
     }
 
     static TestManipulationModel align(File projectRoot, String projectDirName, boolean expectFailure)
             throws IOException, URISyntaxException {
 
+        return align(projectRoot, projectDirName, expectFailure, new HashMap<>());
+    }
+
+    static TestManipulationModel align(File projectRoot, String projectDirName, boolean expectFailure,
+            Map<String, String> systemProps)
+            throws IOException, URISyntaxException {
+
         FileUtils.copyDirectory(Paths
                 .get(TestUtils.class.getClassLoader().getResource(projectDirName).toURI()).toFile(), projectRoot);
-        return align(projectRoot, expectFailure);
+        return align(projectRoot, expectFailure, systemProps);
     }
 
     /**
      * this method assumes the projectRoot directory already contains the gradle files (usually unpacked from resources)
      * 
      * @param projectRoot the root directory of the aligned project
      * @param expectFailure if the the alignment should fail
+     * @param systemProps the system properties to apply for the alignment run
      * @return the manipulation model
      */
-    static TestManipulationModel align(File projectRoot, boolean expectFailure) {
+    static TestManipulationModel align(File projectRoot, boolean expectFailure, Map<String, String> systemProps) {
         assertThat(projectRoot.toPath().resolve("build.gradle")).exists();
 
         final BuildResult buildResult;
         final TaskOutcome outcome;
 
+        final Map<String, String> finalSystemProps = new HashMap<>(systemProps);
+        if (!finalSystemProps.containsKey("repoRemovalBackup")) {
+            finalSystemProps.put("repoRemovalBackup", "settings.xml");
+        }
```

Comment:
I don't think this check is needed? Won't later arguments override earlier?

Reply:
The idea was to add `-DrepoRemovalBackup=settings.xml` as a default value. If users don't supply a value for `repoRemovalBackup` then `settings.xml` should be used, otherwise their value take precedence.

Reply:
Their value would take precedence anyway as its later in the -D list ?

Reply:
Aha, now I see what you are saying